### ISSUE TITLE
TensorRT-8.6.0.12：onnx to tensorrt error：Assertion `!transp_src_ten->is_mod()' failed.

### ISSUE THREAD

**Issue by chenpaopao** [NVIDIA/TensorRT#2873]
## Description

When I try to convert onnx to tensorrt, I get an error：
[04/13/2023-12:10:20] [V] [TRT] Tactic: -504296718212024303 Time: 0.046208
[04/13/2023-12:10:20] [V] [TRT] Fastest Tactic: 5108195418077987197 Time: 0.01984
[04/13/2023-12:10:20] [V] [TRT] >>>>>>>>>>>>>>> Chose Runner Type: CaskConvolution Tactic: 5108195418077987197
[04/13/2023-12:10:20] [V] [TRT] =============== Computing costs for 
[04/13/2023-12:10:20] [V] [TRT] *************** Autotuning format combination: Float(8192,64,8,1) -> Float(2048,16,4,1) ***************
[04/13/2023-12:10:20] [V] [TRT] --------------- Timing Runner: AveragePool_76 (TiledPooling)
[04/13/2023-12:10:20] [V] [TRT] Tactic: 7405825 Time: 0.098688
[04/13/2023-12:10:20] [V] [TRT] Tactic: 7405826 Time: 0.05568
[04/13/2023-12:10:20] [V] [TRT] Tactic: 7405828 Time: 0.034304
[04/13/2023-12:10:20] [V] [TRT] Tactic: 7406081 Time: 0.055552
[04/13/2023-12:10:20] [V] [TRT] Tactic: 7406082 Time: 0.034304
[04/13/2023-12:10:20] [V] [TRT] Tactic: 7406084 Time: 0.023168
[04/13/2023-12:10:20] [V] [TRT] Tactic: 7406337 Time: 0.055296
[04/13/2023-12:10:20] [V] [TRT] Tactic: 7406338 Time: 0.034176
[04/13/2023-12:10:20] [V] [TRT] Tactic: 7406340 Time: 0.022784
[04/13/2023-12:10:20] [V] [TRT] Tactic: 7406593 Time: 0.034304
[04/13/2023-12:10:20] [V] [TRT] Tactic: 7406594 Time: 0.022912
[04/13/2023-12:10:20] [V] [TRT] Tactic: 7406596 Time: 0.017664
[04/13/2023-12:10:20] [V] [TRT] Fastest Tactic: 7406596 Time: 0.017664
[04/13/2023-12:10:20] [V] [TRT] --------------- Timing Runner: AveragePool_76 (CudnnPooling)
[04/13/2023-12:10:20] [V] [TRT] Tactic: -1 Time: 0.014592
[04/13/2023-12:10:20] [V] [TRT] Fastest Tactic: -1 Time: 0.014592
[04/13/2023-12:10:20] [V] [TRT] >>>>>>>>>>>>>>> Chose Runner Type: CudnnPooling Tactic: -1
[04/13/2023-12:10:20] [V] [TRT] =============== Computing costs for 
[04/13/2023-12:10:20] [V] [TRT] *************** Autotuning format combination: Float(2048,16,4,1), Float(786432,262144,512,1), Float(262144,262144,512,1) -> Float(32,1), Float(1310720,262144,512,1) ***************
[04/13/2023-12:10:20] [V] [TRT] --------------- Timing Runner: {ForeignNode[dpn.self_attention.wQs.0.bias + (Unnamed Layer* 56) [Shuffle]...Reshape_186]} (Myelin)
trtexec: /root/gpgpu/MachineLearning/myelin/src/compiler/optimizer/tactic.cpp:1846: bool myelin::ir::tactic_selection_t::global_tactic_constraint_analysis(): Assertion `!transp_src_ten->is_mod()' failed.
Aborted (core dumped)

can you help me to solve this problem？

## Environment

**TensorRT Version**：TensorRT-8.6.0.12、TensorRT-8.5 TensorRT-8.4
**NVIDIA GPU**: 4090
**NVIDIA Driver Version**: 11.8
**CUDA Version**: 11.7
**CUDNN Version**: 
**Operating System**: win11
**Python Version (if applicable)**:python3 
**Tensorflow Version (if applicable)**: 
**PyTorch Version (if applicable)**: 1.11
**Baremetal or Container (if so, version)**: 

## Relevant Files
Onnx file is：

link：https://pan.baidu.com/s/1ZklGjYnMg7-1itmeC6Q4YQ?pwd=fvd5 
code：fvd5 
and
link：https://pan.baidu.com/s/183WefNJlLBMezH2CWh0s4Q?pwd=lryi 
code：lryi 

## Steps To Reproduce
Excuting an order：
trtexec --onnx=model_smi.onnx --saveEngine=out.engine --verbose --workspace=20000
and the error  in linux and win11：
[04/13/2023-12:10:20] [V] [TRT] --------------- Timing Runner: {ForeignNode[dpn.self_attention.wQs.0.bias + (Unnamed Layer* 56) [Shuffle]...Reshape_186]} (Myelin)
trtexec: /root/gpgpu/MachineLearning/myelin/src/compiler/optimizer/tactic.cpp:1846: bool myelin::ir::tactic_selection_t::global_tactic_constraint_analysis(): Assertion `!transp_src_ten->is_mod()' failed.

**Comment by chenpaopao** [comment]
The onnx model can be downloaded here:  https://drive.google.com/drive/folders/1uQX3Clqxhbap_3XDBogDIYIVQniw0JKS?usp=sharing

Please let me know if you have any other questions, I hope you can help me solve this problem

**Comment by zerollzeng** [comment]
```
[04/15/2023-00:52:15] [I] === Performance summary ===
[04/15/2023-00:52:15] [I] Throughput: 3.71826 qps
[04/15/2023-00:52:15] [I] Latency: min = 253.262 ms, max = 259.802 ms, mean = 256.655 ms, median = 257.229 ms, percentile(90%) = 259.219 ms, percentile(95%) = 259.802 ms, percentile(99%) = 259.802 ms
[04/15/2023-00:52:15] [I] Enqueue Time: min = 252.176 ms, max = 259.757 ms, mean = 255.407 ms, median = 255.79 ms, percentile(90%) = 258.209 ms, percentile(95%) = 259.757 ms, percentile(99%) = 259.757 ms
[04/15/2023-00:52:15] [I] H2D Latency: min = 0.206787 ms, max = 0.262329 ms, mean = 0.230737 ms, median = 0.230225 ms, percentile(90%) = 0.253723 ms, percentile(95%) = 0.262329 ms, percentile(99%) = 0.262329 ms
[04/15/2023-00:52:15] [I] GPU Compute Time: min = 252.89 ms, max = 259.413 ms, mean = 256.291 ms, median = 256.889 ms, percentile(90%) = 258.852 ms, percentile(95%) = 259.413 ms, percentile(99%) = 259.413 ms
[04/15/2023-00:52:15] [I] D2H Latency: min = 0.126221 ms, max = 0.135315 ms, mean = 0.133514 ms, median = 0.134094 ms, percentile(90%) = 0.134766 ms, percentile(95%) = 0.135315 ms, percentile(99%) = 0.135315 ms
[04/15/2023-00:52:15] [I] Total Host Walltime: 3.7652 s
[04/15/2023-00:52:15] [I] Total GPU Compute Time: 3.58808 s
[04/15/2023-00:52:15] [W] * Throughput may be bound by Enqueue Time rather than GPU Compute and the GPU may be under-utilized.
[04/15/2023-00:52:15] [W]   If not already in use, --useCudaGraph (utilize CUDA graphs where possible) may increase the throughput.
[04/15/2023-00:52:15] [I] Explanations of the performance metrics are printed in the verbose logs.
[04/15/2023-00:52:15] [V]
[04/15/2023-00:52:15] [V] === Explanations of the performance metrics ===
[04/15/2023-00:52:15] [V] Total Host Walltime: the host walltime from when the first query (after warmups) is enqueued to when the last query is completed.
[04/15/2023-00:52:15] [V] GPU Compute Time: the GPU latency to execute the kernels for a query.
[04/15/2023-00:52:15] [V] Total GPU Compute Time: the summation of the GPU Compute Time of all the queries. If this is significantly shorter than Total Host Walltime, the GPU may be under-utilized because of host-side overheads or data transfers.
[04/15/2023-00:52:15] [V] Throughput: the observed throughput computed by dividing the number of queries by the Total Host Walltime. If this is significantly lower than the reciprocal of GPU Compute Time, the GPU may be under-utilized because of host-side overheads or data transfers.
[04/15/2023-00:52:15] [V] Enqueue Time: the host latency to enqueue a query. If this is longer than GPU Compute Time, the GPU may be under-utilized.
[04/15/2023-00:52:15] [V] H2D Latency: the latency for host-to-device data transfers for input tensors of a single query.
[04/15/2023-00:52:15] [V] D2H Latency: the latency for device-to-host data transfers for output tensors of a single query.
[04/15/2023-00:52:15] [V] Latency: the summation of H2D Latency, GPU Compute Time, and D2H Latency. This is the latency to infer a single query.
[04/15/2023-00:52:15] [I]
&&&& PASSED TensorRT.trtexec [TensorRT v8601] # trtexec --onnx=model_smi.onnx --verbose
```

Looks like this is fixed in the next release, please stay tune :)

**Comment by chenpaopao** [comment]
ok,thanks,looking forword your update. Could you tell the reason for the issuse?

**Comment by zerollzeng** [comment]
It's also a "fix unknown" to me.

**Comment by zerollzeng** [comment]
I just tested with my latest build and found it's fixed.

**Comment by chenpaopao** [comment]
Thanks, can you tell me if your "latest build" version refers to the TensorRT 8.6 EA version in https://developer.nvidia.com/nvidia-tensorrt-8x-download, or the version built according to the guidelines in github.  your build  is in window or linux?

**Comment by zerollzeng** [comment]
Linux, it's an internal build which hasn't released yet. you should see the fix in the next 8.6 GA release.

**Comment by zerollzeng** [comment]
There is no ETA now, and sorry I can't leak any information about it :-）

**Comment by ttyio** [comment]
Closing since 8.6 GA is released, thanks!
